runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -272,6 +273,7 @@ void init_ofiFabricDomain(void) {
 
   hints->caps = FI_MSG | FI_SEND | FI_RECV | FI_MULTI_RECV
                 | FI_RMA | FI_READ | FI_WRITE
+                | FI_ATOMICS
```

Comment:
What happens if atomics aren't available. Does this preclude us from using a provider without atomics, or do we later detect that atomics aren't available and implement them ourselves with am+processors atomics?

Reply:
If atomics or any other thing we've requested are completely unavailable then `fi_getinfo()` should return `-FI_ENODATA` to indicate no matches.  Then we'd have to remove capabilities from the hints until we could get a match.  When our ability to do that ran out we'd have to bail completely.  So far, the providers I've been using (sockets and gni) both support atomics in general and most of the operation/type combinations we use.

Reply:
Hmm ok. Does test/runtime/configMatters/comm/atomics-api*.chpl pass?

I'm assuming things like IBV don't have atomics or at least not all the operations we require. I think we'll eventually need to be smarter here and fallback to our AM+processor atomic implementation. Maybe make an issue for that?

Reply:
`test/runtime/configMatters/comm/*atomics*.chpl` all pass.  We can certainly arrange to handle providers with no atomics support, and it would be easy to drop back to the internal AM+cpuAMO solution if the provider doesn't support atomics at all (versus not supporting certain individual AMO op+type combos).  I'll make an issue for that.  I don't want to code for it right now, though, because I can't test it; we'd have to be able to launch multi-node libfabric-based programs on non-Cray systems to try out the verbs provider for example (see #11116).

Reply:
Late-breaking info: the verbs provider man page says it actually does support `FI_ATOMIC`, but only for `FI_EP_MSG` fabric communication (reliable transfer between connected endpoints).  The ofi comm layer is currently coded for `FI_EP_RDM` fabric communication (reliable transfer among unconnected endpoints).  I don't know how much (re)coding would be needed to allow the comm layer to normally use `FI_EP_RDM` endpoints but drop back to `FI_EP_MSG` endpoints with the verbs provider in order to have `FI_ATOMIC` support, but it could be substantial.  I also don't know how much the `FI_ATOMIC` support would improve performance.